### ISSUE TITLE
Support for inline form evaluation + watching

### ISSUE THREAD

**Issue by mdedetrich** [cursive-ide/cursive#391]
Couldn't find an issue on this, but one of the (only) reasons for using an IDE like lighttable over cursive is its ability to evaluate clojure forms and display the results inline.

An even greater feature is the ability to watch certain forms, and whenever those forms are executed, you see the results of every `s-expression` that gets executed in those forms (this basically completely replaces debugging)

Here is a tutorial video for the evaluate expression
https://www.youtube.com/watch?v=7YIaHyTdjTY
And this one is for the watching feature
https://www.youtube.com/watch?v=d8-b6QEN-rk (note that this is for `javascript`, but it works the exact same way with `clojure`)

Are features like this expected to be integrated into cursive?

**Comment by danielcompton** [comment]
Relates to (dupe of?) #133.

**Comment by cursiveclojure** [comment]
To update this issue - IntelliJ 14 added support for inline painting of results. However the problem is that the results can only be a single line, and there's currently no good way to make them clickable to pop up an inspector window or anything like that. I'm discussing this with JetBrains, and I hope there'll be a good solution in IntelliJ 15.

**Comment by mdedetrich** [comment]
Cool, thanks for looking into this!

**Comment by liango2** [comment]
Yes, I need this ability of evaluation + watching.

**Comment by mdedetrich** [comment]
@cursiveclojure Any idea if this is easier with Intellij 15?

**Comment by mikeball** [comment]
Still interested in this feature, any updates?

**Comment by pradeepcheers** [comment]
@cursive-ide the images in the documentation show that there is an option for inline evaluation. I tried ``CMD + ENTER`` at the end of the line but it doesn't seem to work :(
![image](https://cloud.githubusercontent.com/assets/878734/20928590/8293e6f0-bbbe-11e6-92b5-dd69a1db0cd5.png)

**Comment by danielcompton** [comment]
@pradeepcheers those are just static comments that show what the answer would be, there is no inline evaluation capability yet.

**Comment by maksimr** [comment]
>To update this issue - IntelliJ 14 added support for inline painting of results. However the problem is that the results can only be a single line

For me, it will be enough.
This thing really boosts productivity.

There already exists [plugin](wallabyjs.com) which supports such thing for javascript's tests
<img width="1434" alt="wallaby-js" src="https://cloud.githubusercontent.com/assets/616193/23629615/456d3a4a-02c9-11e7-8667-1619c05968c7.png">

**Comment by cursive-ide** [comment]
@maksimr Yes, I love Wallaby! It's a fantastic tool.

I'm going to try adding this soon to see how it works. My plan is to collapse collection results to a line, truncated if necessary. In the case where a structure is truncated, I'll provide a gutter icon you can click to get a data structure explorer.

**Comment by theronic** [comment]
@cursive-ide please add this! :) Light Table left a gap in the market. I just want to quickly see the results of my thinking inline.

**Comment by alexisvincent** [comment]
Any progress on this? :)

**Comment by coold** [comment]
@cursive-ide any updates?

**Comment by theronic** [comment]
I would love love love to see this in Cursive, because I keep going back to Light Table to debug tricky algorithms and to teach Clojure.

I know Cursive is committed to IntelliJ right now, but it would be worth it to render the evaluated with inline results in a separate window, and in the long-term you'll probably need to do your own rendering.

**Comment by theronic** [comment]
The Calva extension for Visual Studio Code shows the result of the entire expression as an inline comment after the code, but this totally misses the point of debugging intermediate steps, which is what Light Table lets you do, and greatly aids understanding in long threaded pipelines.

Here is a (crappy) macro I wrote to print intermediate results of a let-binding to logs to achieve something similar (using Specter :as S):

```
(defmacro let-spy
  [bindings & body]
  (let [pairs   (partition 2 bindings)
        rebound (into [] cat (S/transform [S/ALL (S/nthpath 1)] #(list 'log/spy :debug %) pairs))]
    `(let ~rebound ~@body)))
```

Surely this should be possible to do in the IDE given that Cursive already has to parse every expression?

**Comment by muhajirdev** [comment]
any update about this?

**Comment by classicrob** [comment]
I'm still curious about this! I'd love it.

**Comment by cursive-ide** [comment]
This is actually the next big new feature I'm working on, but it's stalled momentarily while I have to look at some other things. But, it's on the way.

**Comment by cursive-ide** [comment]
Fix released in 1.14.0-eap1 on 2024-09-13.

**Comment by galemsky-hs** [comment]
@cursive-ide as far as I see, the inline print/result display were implemented, but not the watches, correct?

**Comment by cursive-ide** [comment]
Fix released in 1.14.0 on 2024-11-13.
